server/pulp/server/lazy/streamer.py

```diff
@@ -0,0 +1,212 @@
+from gettext import gettext as _
+from httplib import NOT_FOUND, INTERNAL_SERVER_ERROR
+from urlparse import urlparse
+import logging
+
+from mongoengine import DoesNotExist, MultipleObjectsReturned
+from nectar import listener as nectar_listener
+from nectar import request as nectar_request
+from twisted.internet import reactor
+from twisted.web import resource
+from twisted.web.server import NOT_DONE_YET
+
+from pulp.server.config import config as pulp_config
+from pulp.server.db import model
+from pulp.server.controllers import repository as repo_controller
+from pulp.server.controllers import content as content_controller
+from pulp.plugins.loader.exceptions import PluginNotFound
+
+logger = logging.getLogger(__name__)
+
+
+class StreamerNectarListener(nectar_listener.DownloadEventListener):
```

Comment:
Haha, I never refactored this from the example streamer that was whipped up during pulpcoooooooon. It's a pretty silly name and I will change it.